publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java

```diff
@@ -1294,6 +1295,40 @@ void shouldSetResourceEventToNull() throws BadRequestException, NotFoundExceptio
         assertNull(resource.getResourceEvent());
     }
 
+    @Test
+    void shouldDoNothingWhenPublishingPublicationWhenPublicationIsAlreadyPublished() throws ApiGatewayException {
+        var publication = randomPublication();
+        var userInstance = UserInstance.fromPublication(publication);
+        var peristedPublication = Resource.fromPublication(publication)
+                                      .persistNew(resourceService, userInstance);
+        resourceService.publishPublication(userInstance, peristedPublication.getIdentifier());
+
+        var publishedResource = Resource.fromPublication(peristedPublication).fetch(resourceService);
+
+        Resource.resourceQueryObject(peristedPublication.getIdentifier()).publish(resourceService, userInstance);
+
+        var resource = Resource.fromPublication(peristedPublication).fetch(resourceService);
+
+        assertEquals(PUBLISHED, publishedResource.getStatus());
+        assertEquals(publishedResource.getPublishedDate(), resource.getPublishedDate());
```

Comment:
You might want to`verify` that some method of resource service is not called, rather than just side effects. The first assertion is a bit pointless. Also, you have modified date, I guess?